internal/efitest/vars.go

```diff
@@ -0,0 +1,103 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package efitest
+
+import (
+	"bytes"
+	_ "crypto/sha256"
+	"io"
+
+	efi "github.com/canonical/go-efilib"
+
+	. "gopkg.in/check.v1"
+)
+
+// VarEntry describes the contents of a mock EFI variable.
+type VarEntry struct {
+	Attrs   efi.VariableAttributes
+	Payload []byte
+}
+
+type VarPayloadWriter interface {
+	Write(w io.Writer) error
```

Comment:
nitpick: this is a naming issue not here but in the efi package itself, but usually methods writing to a io.Writer are called WriteTo and not just Write

Reply:
I've used this naming convention in a few packages - I did try adjusting it to `WriteTo` in go-efilib at one point and IIRC it triggered go vet failures because it checks that functions called `WriteTo` and which take a `io.Writer` exactly match the `io.WriterTo` interface, which would mean adjusting them all to return `(int64, error)` as well.